src/core/load-fragment.c

```diff
@@ -3137,14 +3138,23 @@ int config_parse_restrict_namespaces(
                 rvalue++;
         }
 
-        r = parse_restrict_namespaces(rvalue, &c->restrict_namespaces);
+        r = parse_restrict_namespaces(rvalue, &flags);
         if (r < 0) {
                 log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse namespace type string, ignoring: %s", rvalue);
                 return 0;
         }
 
         if (invert)
-                c->restrict_namespaces = (~c->restrict_namespaces) & NAMESPACE_FLAGS_ALL;
+                flags = (~flags) & NAMESPACE_FLAGS_ALL;
+
+        if (c->restrict_namespaces == NAMESPACE_FLAGS_ALL)
+                        c->restrict_namespaces = flags;
+        else {
+                if (invert)
+                        c->restrict_namespaces &= flags;
+                else
+                        c->restrict_namespaces |= flags;
+        }
```

Comment:
I think this is confusing. If I have
```
RestrictNamespaces=cgroup ipc net mnt pid user uts
RestrictNamespaces=cgroup
```
the result is `RestrictNamespaces=cgroup`.
But if I have 
```
RestrictNamespaces=ipc net mnt pid user uts
RestrictNamespaces=cgroup
```
the result is `RestrictNamespaces=cgroup ipc net mnt pid user uts`.

I think that's just too confusing.

Reply:
I expect that the both two examples in your comment result `RestrictNamespaces=cgroup ipc net mnt pid user uts`. In the revised commits, I've added examples in the man page.

Reply:
@yuwata the prob is that `c->restrict_namespaces == NAMESPACE_FLAGS_ALL` will hold if the user explicitly enabled all namespaces first... i mean NAMESPACE_FLAGS_ALL is after all the mask where all controllers are on...